src/main/java/de/yuna/berlin/nativeapp/services/http/model/HttpRequest.java

```diff
@@ -0,0 +1,250 @@
+package de.yuna.berlin.nativeapp.services.http.model;
+
+import berlin.yuna.typemap.logic.JsonDecoder;
+import berlin.yuna.typemap.model.TypeContainer;
+import berlin.yuna.typemap.model.TypeMap;
+import com.sun.net.httpserver.Headers;
+import com.sun.net.httpserver.HttpExchange;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class HttpRequest {
+
+    protected final HttpMethod method;
+    protected final String path;
+    protected final TypeMap headers;
+    protected final HttpExchange exchange;
+    protected String body;
+    protected TypeMap queryParams;
+    protected TypeMap pathParams;
+
+    public HttpRequest(HttpExchange exchange) {
+        this.method = HttpMethod.httpMethodOf(exchange.getRequestMethod());
+        this.path = exchange.getRequestURI().getPath();
+        this.headers = convertHeaders(exchange.getRequestHeaders());
+        this.exchange = exchange;
+    }
+
+    public HttpMethod getMethod() {
```

Comment:
Please no old school `get` or `set` as it breaks the nature of chaining and fluent methods. Just call it `method()`.